app.py

```diff
@@ -79,13 +79,14 @@ async def echo(message, history, state, persona):
         )        
     
     # make summary
+    use_generated_summaries = True if use_generated_summaries == "Yes" else False
     response = await client.models.generate_content(
         model=args.model,
         contents=[
             Template(
                 prompt_tmpl['summarization']['prompt']
             ).safe_substitute(
-                previous_summary=state['summary'], 
+                previous_summary=state['summary'] if use_generated_summaries else "No previous summary", 
```

Comment:
I don't think this line needs to be changed. 
Instead, we need to change line number 61 because we want to see how the model's response changes based on the summary or the entire conversation.